src/main/groovy/org/agileworks/elasticsearch/river/csv/processrunner/ProcessRunnerFactory.groovy

```diff
@@ -0,0 +1,21 @@
+package org.agileworks.elasticsearch.river.csv.processrunner
+
+import groovy.transform.CompileStatic
+
+@CompileStatic
+class ProcessRunnerFactory {
+
+    private static final ProcessRunnerFactory INSTANCE = new ProcessRunnerFactory()
```

Comment:
I previously tried this, but I could not get ProcessRunnerFactory.getInstance().getRunner() to work in CSVRiver.java that way.
Maven didn't report the actual error, even with -e and -X switches, but it did not compile.
@Singleton in the groovy code did work though.